### PR TITLE
[AlarmDecoder] Added support for AlarmDecoder LRR Protocol

### PR DESCRIPTION
**Pull request by LuckyMallari**
Overview of LRR Support : http://www.alarmdecoder.com/wiki/index.php/LRR_Support
Alarm Decoder LRR Protocol : http://www.alarmdecoder.com/wiki/index.php/Protocol#LRR

Sample usage:

Item MUST be in this format:
**String sAlarmLRR "LLR Msg: [%s]" (gAlarmPanel) {alarmdecoder="LRR:00#text"}**

ITEMS:
```
String sAlarmLRR "LLR Msg: [%s]" (gAlarmPanel) {alarmdecoder="LRR:00#text"}
String sAlarmLRR_lastUser "Last User: [%s]" (gAlarmPanel)
String sAlarmLRR_lastEvent "Last Event: [%s]" (gAlarmPanel)
DateTime sAlarmLRR_lastEventDateTime ""  (gAlarmPanel)
String sAlarmLRR_lastPartition "Last Partition: [%s]" (gAlarmPanel)
```

RULES:
```

import java.util.Map;
import java.util.HashMap;
import com.google.common.collect.ImmutableMap

val Map<String, String> UserList = ImmutableMap.<String, String>builder()
    .put("001", "Installer")
    .put("002", "Master")
    .put("003", "User1")
    .put("004", "User2")
    .build()

rule "LRR Parser"
when
	Item sAlarmLRR received update
then
	var String msg = sAlarmLRR.state.toString();
	var String[] parts = msg.split(",");
	if (parts.length != 3) {
		logError("rules", "sAlarmLRR rule parsing error.");
	} else {
		logInfo("rules", "Parsing sAlarmLRR==> " + msg);
		var String user = UserList.get(parts.get(0));	
		var String partition = parts.get(1);
		var String event = transform("MAP", "alarm_LRR_eventmap.map", parts.get(2).toString);
		
		postUpdate(sAlarmLRR_lastUser, user);
		postUpdate(sAlarmLRR_lastPartition, partition);
		postUpdate(sAlarmLRR_lastEvent, event);
		postUpdate(sAlarmLRR_lastEventDateTime, new DateTimeType());
	}
end
```

### CODE REVIEW

**Comment by watou** [comment]
Thanks @LuckyMallari!  @berndpfrommer OK to merge or do you have changes to suggest?

**Comment by LuckyMallari** [comment]
@watou There are no breaking changes, so should be OK to just merge

File: bundles/binding/org.openhab.binding.alarmdecoder/src/main/java/org/openhab/binding/alarmdecoder/internal/ADMsgType.java [hunk 1]
```diff
@@ -21,8 +21,9 @@
     RFX(1), // wireless message
     EXP(2), // zone expander message
     REL(3), // relay message
-    INVALID(4), // invalid message
-    NUMTYPES(5); // total number of types
+    LRR(4), // long range radio message
+    INVALID(5), // invalid message
+    NUMTYPES(7); // total number of types
```
Comment (watou) [inline bundles/binding/org.openhab.binding.alarmdecoder/src/main/java/org/openhab/binding/alarmdecoder/internal/ADMsgType.java]: I agree with @berndpfrommer's comment:

> why is NUMTYPES initialized to 7? Shouldn't it be 6? Looks like a bug to me.

Comment (LuckyMallari) [inline bundles/binding/org.openhab.binding.alarmdecoder/src/main/java/org/openhab/binding/alarmdecoder/internal/ADMsgType.java]: Fixed. Perhaps a typo as I was trying a custom AlarmDecoder format.

https://github.com/LuckyMallari/openhab1-addons/commit/9c42cd2364efba659ed3f33564a9a3692494caa1

**Comment by watou** [comment]
Thanks @LuckyMallari!  Please update the wiki to include this enhancement, noting its first appearance in the 1.10.0 release.

**Comment by LuckyMallari** [comment]
@watou 

Updated : https://github.com/openhab/openhab1-addons/wiki/AlarmDecoder-binding

**Comment by watou** [comment]
Thank you very much, and thanks for the heads-up as well, since I'm in the midst of converting the wiki pages to README.md files so they end up at docs.openhab.org.  (See https://github.com/watou/openhab/tree/ded066c197e08d251346827e7e1ac77ef2413a5b/bundles/binding/org.openhab.binding.alarmdecoder) where I've tried to capture your new edits.)

**Pull request merged**